src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -177,7 +179,38 @@ void updateSecurityGroupEntity(SecurityGroupPatch securityGroupPatch, SecurityGr
     }
 
     private void patchSecurityGroupUsers(SecurityGroupPatch securityGroupPatch, SecurityGroupEntity securityGroupEntity) {
+
         List<Integer> userIds = securityGroupPatch.getUserIds();
+        if (userIds != null) {
+            // get a list of system users for security group
+            List<Integer> systemUserIds = securityGroupEntity
+                .getUsers()
+                .stream()
+                .filter(user -> user.getIsSystemUser())
+                .map(UserAccountEntity::getId)
+                .toList();
+
+            // filter the incoming list to remove system users
+            List<UserAccountEntity> patchUsers = userAccountRepository
+                .findByIdIn(securityGroupPatch.getUserIds());
+
+            if (userIds.size() > 0 && patchUsers.isEmpty()) {
+                throw new DartsApiException(
+                    UserManagementError.USER_NOT_FOUND,
+                    String.format("No User accounts found for patch user IDs %s", securityGroupPatch.getUserIds()));
+            }
+
+            List<Integer> patchNonSystemUserIds = patchUsers
+                .stream().filter(user -> !user.getIsSystemUser())
+                .map(UserAccountEntity::getId)
+                .toList();
+
+            // join the 2 lists - distinct
+            userIds = Stream.concat(systemUserIds.stream(), patchNonSystemUserIds.stream())
+                .distinct()
+                .collect(Collectors.toList());
+        }
+
         if (userIds != null) {
```

Comment:
The condtional on L214 duplicates the conditonal on L184, so can we just have a single conditional block for all this code?

Or it may be even cleaner to just return early if userIds is null, and avoid a layer of nesting altogether. E.g:

```java
    private void patchSecurityGroupUsers(SecurityGroupPatch securityGroupPatch, SecurityGroupEntity securityGroupEntity) {

        List<Integer> userIds = securityGroupPatch.getUserIds();
        if (userIds == null) {
            return;
        }

       // All the meat of the existing method goes here.
```

Reply:
yeh. i think the confusion is in the re-use of userIds later on for the combined list. I've changed to return if userIds is initially null - and used a different name (combinedUserIds) for the combined list